enterprise/ha/src/main/java/org/neo4j/kernel/ha/cluster/SwitchToSlave.java

```diff
@@ -425,17 +438,14 @@ private URI startHaCommunication( LifeSupport haCommunicationLife, NeoStoreDataS
         return slaveHaURI;
     }
 
-    private boolean catchUpWithMaster()
+    private boolean catchUpWithMaster(UpdatePuller updatePuller)
             throws IllegalArgumentException, InterruptedException
     {
         monitor.catchupStarted();
         RequestContext catchUpRequestContext = requestContextFactory.newRequestContext();
         console.log( "Catching up with master. I'm at " + catchUpRequestContext );
 
-        // Unpause the update puller, because we know that we are a slave that just started communication with master.
-        UpdatePuller updatePuller = resolver.resolveDependency( UpdatePuller.class );
-        updatePuller.unpause();
-        if ( !updatePuller.await( UpdatePuller.NEXT_TICKET, false ) )
+        if ( !updatePuller.tryPullUpdates() )
```

Comment:
this `if` was added as a tentative fix (which is not really working, hence this PR), can we revert back to the original implementation, i.e., by pulling updates no matter what, or is there a different reason why this might fail?